Sources/Networking/v2/APIService.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  APIService.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import os.log
+
+public protocol APIService {
+
+    typealias APIResponse = (data: Data?, httpResponse: HTTPURLResponse)
+
+    func fetch<T: Decodable>(request: APIRequestV2) async throws -> T?
+    func fetch(request: APIRequestV2) async throws -> APIService.APIResponse
+}
+
+public struct DefaultAPIService: APIService {
+    private let urlSession: URLSession
+
+    public init(urlSession: URLSession = .shared) {
```

Comment:
We should consider also injecting the `JSONDecoder`. This might be useful if we want to change decoding attributes. e.g for APIs with snake case:

```
let decoder = JSONDecoder()
decoder.keyDecodingStrategy = .convertFromSnakeCase
```

Reply:
+1 for that suggestion, snake case and date time formats are prime examples where it may be handy.